pkg/runtask/v1beta1/kubernetes_client.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/runtask/v1beta1"
+	"github.com/openebs/maya/pkg/client/generated/openebs.io/runtask/v1beta1/clientset/internalclientset"
+	kube "github.com/openebs/maya/pkg/kubernetes/client/v1alpha1"
+	"github.com/pkg/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"strings"
+)
+
+// getClientsetFunc is a typed function to get runtask's clientset
+//
+// NOTE:
+//  functional makes it simple to mock
+type getClientsetFunc func(*rest.Config) (*internalclientset.Clientset, error)
+
+// getClientset returns a new instance of runtask based clientset
+//
+// NOTE:
+//  This is an implementation of getClientsetFunc
+func getClientset(c *rest.Config) (*internalclientset.Clientset, error) {
+	return internalclientset.NewForConfig(c)
+}
+
+// getRuntaskFunc is a typed function to get RunTask instance
+//
+// NOTE:
+//  functional makes it simple to mock
+type getRuntaskFunc func(k *kubeclient, cs *internalclientset.Clientset, name string) (*apis.RunTask, error)
+
+// getRuntask returns the Runtask instance corresponding to
+// the provided name
+//
+// NOTE:
+//  This is an implementation of getRuntaskFunc
+func getRuntask(k *kubeclient, cs *internalclientset.Clientset, name string) (*apis.RunTask, error) {
+	if k == nil {
+		return nil, errors.New("failed to get runtask: nil kubeclient was provided")
+	}
+	if cs == nil {
+		return nil, errors.New("failed to get runtask: nil clientset was provided")
+	}
+	return cs.OpenebsV1beta1().RunTasks(k.namespace).Get(name, v1.GetOptions{})
+}
+
```

Comment:
For the above -- what is resource is clustered scope ? In this case, we do not require `k` .
Why both `kubeclient` and `internalclient` are used here ?